### PR TITLE
61 update datawolf execution step dynamically

### LINKED ISSUE #61: list datawolf execution step progress and use their output when it's ready
each analysis is now separated into different steps
we should be able to on the frontend render the step status and show the user their results along the way

### PR DESCRIPTION
**Pull request by Rashmil-1999**
This PR enables the results to be viewed as soon as the required output datasets are ready after execution from datawolf workflows. To test it, 
- I used SLC playbook's datawolf configuration and ran an analysis in step 3-4. 
- Toggle the running task to view results.
- Wait for the link to be clickable as soon as the requirements are met automatically.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 	"name": "@in-core/incore-playbook",
-	"version": "0.4.4",
+	"version": "0.4.5",
 	"lockfileVersion": 2,
 	"requires": true,
 	"packages": {
 		"": {
 			"name": "@in-core/incore-playbook",
-			"version": "0.4.4",
+			"version": "0.4.5",
```
Comment (longshuicy) [inline package-lock.json]: I published 0.4.5 already maybe you can bump everything from now on to 0.4.6

File: src/components/childComponents/table/ViewResultEntryTable.jsx [hunk 2]
```diff
@@ -166,10 +166,23 @@ export function ViewResultEntryTable(props) {
 		handleRequestSort(event, property);
 	};
 
+	const checkResultsReady = (listWidgets, currentExecutionDatasetIds) => {
```
Comment (longshuicy) [inline src/components/childComponents/table/ViewResultEntryTable.jsx]: very elegant!

File: src/components/childComponents/table/ViewResultEntryTable.jsx [hunk 3]
```diff
@@ -166,10 +166,23 @@ export function ViewResultEntryTable(props) {
...
+		const requiredDatasetIds = datawolfWorkflowConfig.outputs.datasets[0].datasetsMap.reduce((list, item) => {
+			if (listWidgets.includes(item.datasets.widget)) {
+				return [...list, item.datasetId];
+			}
+
+			return list;
+		}, []);
+
+		return requiredDatasetIds.every((reqId) => currentExecutionDatasetIds.includes(reqId));
```
Comment (longshuicy) [inline src/components/childComponents/table/ViewResultEntryTable.jsx]: Sometimes when an step failed, the dataset will display as below
`"0105b539-d201-4eae-f66e-be3fb05d7683": "ERROR",`

SO maybe it's worth to pass the whole dataset object in `execution.datasets` and add one additional check 
```
&& execution.datasets[reqId] !== "ERROR"
```

**Review by longshuicy — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (Rashmil-1999) [reply package-lock.json]: So, is it okay if I do that in the next PR which is pertaining to the date time format beautification? or I can do this here and keep the same version in the next PR.

File: src/components/childComponents/table/ViewResultEntryTable.jsx (hunk 3 above)
Comment (Rashmil-1999) [reply src/components/childComponents/table/ViewResultEntryTable.jsx]: Okay, got it. I thought "what is the value of the dataset id, is it important or not?" but then just went with the simple approach... Thank you for clearing that up!

Comment (longshuicy) [reply src/components/childComponents/table/ViewResultEntryTable.jsx]: haha sorry i wasn't super clear before

**Comment by longshuicy** [comment]
Could you resolve the merge conflict?

**Review by longshuicy — APPROVED** [review]

**Comment by longshuicy** [comment]
Sorry you have to resolve merge conflict again cuz I merged your other PR in

**Pull request merged**